seabird/modules/bleep.py

```diff
@@ -0,0 +1,109 @@
+from functools import lru_cache
+import random
+import re
+
+from sqlalchemy import Column, String
+from sqlalchemy.exc import IntegrityError
+
+from seabird.plugin import Plugin, CommandMixin
+
+from .db import Base, DatabaseMixin
+
+
+class Bleep(Base):
+    __tablename__ = 'bleep'
+
+    bad_word = Column(String, primary_key=True)
+    replacement = Column(String)
+
+
+class BleepPlugin(Plugin, CommandMixin, DatabaseMixin):
+    REPLIES = [
+        'Hey, watch your mouth! Say "{}" instead.',
+        'Pottymouth! We say "{}" in this channel.',
+        'Uh oh, you should really say "{}" instead.',
+        'Time to put a quarter in the jar! You should really use "{}" instead.',
+    ]
+
+    @lru_cache(maxsize=1)
+    def _get_bleeps(self):
+        '''
+        Gets a list of bleeped words and their replacements
+
+        @return [Bleep] List of bleeps
+        '''
+        with self.db.session() as session:
+            query = session.query(Bleep)
+
+            bleeps = query.all()
+
+            # This call is necessary to use the retrieved bleeps outside the
+            # scope of the with
+            session.expunge_all()
+
+            return bleeps
+
+    def cmd_bleep(self, msg):
+        '''
+        Begin to bleep `bad_word` with `replacement`.
+
+        `bad_word` is args[0]
+        `replacement` is args[1]
+        '''
+        args = msg.trailing.lower().strip().split(' ')
+
+        if len(args) < 2:
+            self.bot.reply(
+                msg,
+                'Must supply a bad word and a replacement',
+            )
+            return
+
+        bad_word = args[0]
+        replacement = args[1]
+
+        bleep = Bleep()
+        bleep.bad_word = args[0]
+        bleep.replacement = args[1]
+        with self.db.session() as session:
```

Comment:
I specifically implemented this in the DB session so we wouldn't have to keep doing this. There's get_or_create on the session object. I believe it's used in the karma module.